src/oneD/StFlow.cpp

```diff
@@ -141,6 +148,50 @@ void StFlow::setThermo(IdealGasPhase& th) {
     m_thermo = &th;
 }
 
+void StFlow::setKinetics(shared_ptr<Kinetics> kin)
+{
+    if (!m_solution) {
+        // @todo remove after Cantera 3.0
+        throw CanteraError("StFlow::setKinetics",
+            "Unable to update object that was not constructed from smart pointers.");
+    }
+    m_kin = kin.get();
+    m_solution->setKinetics(kin);
+}
+
+void StFlow::setKinetics(Kinetics& kin)
+{
+    // @todo: resolve crash for updated m_solution->registerChangedCallback
+    // warn_deprecated("StFlow::setKinetics",
+    //     "To be removed after Cantera 3.0. Replaced by Domain1D::setKinetics.");
+    m_kin = &kin;
```

Comment:
Ah - yes, this is something that I wanted to come back to, but didn't. The crash I experienced happened when I tried to update https://github.com/Cantera/cantera/blob/7f04d839f3f0bebbc3474ecb972007d896362af7/src/oneD/StFlow.cpp#L126-L129
to used the `shared_ptr` version. It may have masked another issue and I haven't tried since. Will certainly look at it again. The deprecation warning is commented out as it is de facto still actively used by the callbacks in `StFlow` and `IonFlow`.

Reply:
@speth ... I just confirmed that this still crashes: simply remove the `*`'s (either of the two lines), recompile and run `scons test-python-onedim`. Any insights would be appreciated ...